src/fr-device/collector.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @class Collector - base class for FRDevice
+ * Generic collector functions for collecting a device profile attributes
+ */
+class Collector {
+  /**
+   * @method reduceToObject - goes one to two levels into source to collect attribute
+   * @param props - array of strings; can use dot notation for two level lookup
+   * @param src - source of attributes to check
+   */
+  reduceToObject(props: string[], src: any): { [key: string]: string } {
+    return props.reduce((prev, curr) => {
+      if (curr.includes('.')) {
+        const propArr = curr.split('.');
+        const prop = src[propArr[0]] && src[propArr[0]][propArr[1]];
+        prev[propArr[1]] = prop != undefined ? prop : '';
+      } else {
+        prev[curr] = src[curr] != undefined ? src[curr] : null;
```

Comment:
Unfortunately no, and that's because the condition needs to be `false` _only_ in the case of a value being `undefined` or `null`, which `x != undefined` magically does for us. It's important that other falsy values like `''`, `false`, `0` ... result in `true`. Maybe I'll add a comment here explaining that.